### PR TITLE
boards/samr21-xpro: add sx127x driver pin definition 

### PR DESCRIPTION
**Pull request by fjmolinas**
Adds pin configuration to work with sx127x driver.

### CODE REVIEW

File: boards/samr21-xpro/include/board.h [hunk 1]
```diff
@@ -52,6 +52,21 @@ extern "C" {
                                      .reset_pin = GPIO_PIN(PB, 15)}
 
 /**
+ * @name    SX127x configuration
+ *
+ * {spi bus, nss pin, reset pin & dioX pins}
+ */
+
+#define SX127X_PARAMS_BOARD       { .spi       = SPI_DEV(1), \
```
Comment (aabadie) [inline boards/samr21-xpro/include/board.h]: This is not part of the board but can be plugged to it from outside: the right place is in a separate `sx127x_params.h` that can be in `boards/samr21-xpro/include` directory. See `boards/arduino-due` for an example.

This comment is also valid with #7340.

**Comment by aabadie** [comment]
Thinking of it, since there's no LoRa hardware designed for this board (e.g an extension board with Semtech), I don't think this PR is needed.
You could just add an adapted version of `sx127x_params.h` in your application directory without modifying the board code.
Imagine if we generalize your change strategy for all drivers on all boards, the code would become a mess. :)

**Comment by fjmolinas** [comment]
@aabadie So we just drop this one and keep #7340 since it does have a specific LoRa Hw?

**Comment by aabadie** [comment]
> So should we just drop this one and keep #7340 since it does have a specific LoRa Hw?

I would say yes

**Comment by aabadie** [comment]
> yes

for both

**Pull request closed without merging**